services/blockchain-app-registry/tests/functional/shared/metadataIndex.test.js

```diff
@@ -0,0 +1,459 @@
+		expect(dbResponse.length).toEqual(1);
+
+		await deleteIndexedMetadataFromFile(tokenMetaPath);
+
+		const dbResponseAfterDelete = await tokenMetadataTable.find(tokenMetaQueryParams, ['chainID']);
+		expect(dbResponseAfterDelete.length).toEqual(0);
+	});
+
+	it('should delete indexed app meta in db only after commit when called with valid app meta path and dbTrx', async () => {
+		const responseBefore = await applicationMetadataTable.find(
+			{ network: indexAppMetaInput.networkType, chainName: indexAppMetaInput.chainName },
+			['chainID'],
+		);
+		expect(responseBefore[0].chainID).toEqual(indexAppMetaInput.chainID);
+		expect(responseBefore.length).toEqual(1);
+
+		const dbTrx = await startDbTransaction(connection);
+		await deleteIndexedMetadataFromFile(appMetaPath, dbTrx);
+
+		const responseBeforeCommit = await applicationMetadataTable.find(
+			{ network: indexAppMetaInput.networkType, chainName: indexAppMetaInput.chainName },
+			['chainID'],
+		);
+		expect(responseBeforeCommit[0].chainID).toEqual(indexAppMetaInput.chainID);
+		expect(responseBeforeCommit.length).toEqual(1);
+
+		await commitDbTransaction(dbTrx);
+
+		const responseAfterCommit = await applicationMetadataTable.find(
+			{ network: indexAppMetaInput.networkType, chainName: indexAppMetaInput.chainName },
+			['chainID'],
+		);
+		expect(responseAfterCommit.length).toEqual(0);
+	});
+
+	it('should delete indexed token meta in db only after commit when called with valid token meta path and dbTrx', async () => {
+		const responseBefore = await tokenMetadataTable.find(tokenMetaQueryParams, ['chainID']);
+		expect(responseBefore[0].chainID).toEqual(indexAppMetaInput.chainID);
+		expect(responseBefore.length).toEqual(1);
+
+		const dbTrx = await startDbTransaction(connection);
+		await deleteIndexedMetadataFromFile(tokenMetaPath, dbTrx);
+
+		const responseBeforeCommit = await tokenMetadataTable.find(tokenMetaQueryParams, ['chainID']);
+		expect(responseBeforeCommit[0].chainID).toEqual(indexAppMetaInput.chainID);
+		expect(responseBeforeCommit.length).toEqual(1);
+
+		await commitDbTransaction(dbTrx);
+
+		const responseAfterCommit = await tokenMetadataTable.find(tokenMetaQueryParams, ['chainID']);
+		expect(responseAfterCommit.length).toEqual(0);
+	});
+
+	it('should throw error when called with null or undefined filePath', async () => {
+		expect(() => deleteIndexedMetadataFromFile(null)).rejects.toThrow();
+		expect(() => deleteIndexedMetadataFromFile(undefined)).rejects.toThrow();
+	});
+});
+
+describe('Test indexAllBlockchainAppsMeta method', () => {
+	beforeEach(async () => {
+		await rmdir(config.dataDir);
+		await applicationMetadataTable.delete({});
+		await tokenMetadataTable.delete({});
+	});
+
+	it('should throw error when data is not present', async () => {
+		expect(() => indexAllBlockchainAppsMeta()).rejects.toThrow();
+	});
+
+	xit('should index app and token meta in db when data is present', async () => {
```

Comment:
Suppressed due to `octokit` issue of generating `HttpError`

Reply:
This can be probably mocked using Jest. Let's discuss that offline and fix it.